asyncstdlib/functools.py

```diff
@@ -57,6 +58,25 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}({self.value!r})"
 
 
+class _RepeatableCoroutine(Generic[T]):
+    """Helper to ``await`` a coroutine also more or less than just once"""
+
+    __slots__ = ("call", "args", "kwargs")
+
+    def __init__(
+        self, __call: Callable[..., Coroutine[Any, Any, T]], *args: Any, **kwargs: Any
+    ):
+        self.call = __call
+        self.args = args
+        self.kwargs = kwargs
+
+    def __await__(self) -> Generator[None, None, T]:
```

Comment:
This may need to be `Generator[Any, Any, T]`, since an event loop may be using the send/yield API differently than `asyncio`.